inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaClassElement.java

```diff
@@ -796,15 +801,32 @@ protected Collection<TypeElement> getInterfaces(TypeElement classNode) {
         }
 
         @Override
-        protected List<Element> getEnclosedElements(TypeElement classNode, ElementQuery.Result<?> result) {
+        protected List<Element> getEnclosedElements(TypeElement classNode, ElementQuery.Result<?> result, boolean includeAbstract) {
             List<? extends Element> ee;
             if (classNode == classElement) {
                 ee = getEnclosedElements();
             } else {
                 ee = classNode.getEnclosedElements();
             }
             EnumSet<ElementKind> elementKinds = getElementKind(result);
-            return ee.stream().filter(element -> elementKinds.contains(element.getKind())).collect(Collectors.toList());
+            List<Element> list = new ArrayList<>(elementKinds.size());
+            for (Element element : ee) {
+                Set<Modifier> modifiers = element.getModifiers();
+                if (elementKinds.contains(element.getKind()) && (includeAbstract || isNonAbstractMethod(modifiers, classNode))) {
+                    list.add(element);
+                }
+            }
+            return list;
+        }
+
+        private boolean isNonAbstractMethod(Set<Modifier> modifiers, TypeElement classNode) {
+            if (modifiers.contains(Modifier.DEFAULT)) {
+                return true;
+            }
+            if (modifiers.contains(Modifier.PRIVATE) && isInterface(classNode)) {
```

Comment:
here it's the opposite from above? maybe the above version is wrong?

Reply:
why the isInterface check then?

Reply:
Yeah, it's not needed, removed